File: src/main/java/io/github/romeoahmed/cursedoath/domain/DomainSounds.java
package io.github.romeoahmed.cursedoath.domain;

import io.github.romeoahmed.cursedoath.CursedOath;
import net.minecraft.core.Registry;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.sounds.SoundEvent;

public final class DomainSounds {
    private DomainSounds() {}

    public static final SoundEvent OPEN = register("domain_open"),
            CLOSE = register("domain_close"),
            CUT = register("shrine_cut"),
            BREAK = register("barrier_break");

    public static void initialize() {}

    private static SoundEvent register(String path) {
        var id = CursedOath.id(path);
        return Registry.register(BuiltInRegistries.SOUND_EVENT, id, SoundEvent.createVariableRangeEvent(id));
    }
}

File: src/main/kotlin/io/github/romeoahmed/cursedoath/domain/DomainSounds.kt
package io.github.romeoahmed.cursedoath.domain

import io.github.romeoahmed.cursedoath.CursedOath
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.sounds.SoundEvent

object DomainSounds {
    val OPEN = register("domain_open")
    val CLOSE = register("domain_close")
    val CUT = register("shrine_cut")
    val BREAK = register("barrier_break")

    fun initialize() = Unit

    private fun register(path: String): SoundEvent {
        val id = CursedOath.id(path)
        return Registry.register(BuiltInRegistries.SOUND_EVENT, id, SoundEvent.createVariableRangeEvent(id))
    }
}
